samza-core/src/main/java/org/apache/samza/serializers/JsonCheckpoint.java

```diff
@@ -29,6 +29,7 @@
 public class JsonCheckpoint {
   private String checkpointId;
   private Map<String, Map<String, String>> inputOffsets;
+  // Map<StorageBackendFactoryName, Map<StoreName, StateCheckpointMarker>>
```

Comment:
+1, maybe use this convention in javadocs at-params/at-returns and comments describing field/variables as well. More readable than the natural language description IMHO.